apps/examples/testcase/ta_tc/systemio/itc/itc_i2c.c

```diff
@@ -115,15 +116,26 @@ void itc_iotbus_i2c_set_address_p(void)
 void itc_iotbus_i2c_write_read_p(void)
 {
 	int ret = IOTBUS_ERROR_NONE;
-	uint8_t szCmd[2] = { 0x01, 0x02 };
-	uint8_t szReadBuf[10];
+	uint8_t write_cmd[2] = { 0x1, 0x2};
+	uint8_t address = 0x23;
+	uint8_t read_buf[10];
+
 	iotbus_i2c_context_h h_i2c = iotbus_i2c_init(g_bus);
 	TC_ASSERT_NEQ("iotbus_i2c_init", h_i2c, NULL);
 
-	ret = iotbus_i2c_write(h_i2c, szCmd, 1);
-	TC_ASSERT_EQ_CLEANUP("iotbus_i2c_write", (ret < 0), false, iotbus_i2c_stop(h_i2c));
+	ret = iotbus_i2c_set_address(h_i2c, address);
+	TC_ASSERT_EQ_CLEANUP("iotbus_i2c_set_address", ret, IOTBUS_ERROR_NONE, iotbus_i2c_stop(h_i2c));
+
+	ret = iotbus_i2c_write(h_i2c, &write_cmd[0], 1);
+	TC_ASSERT_GT_CLEANUP("iotbus_i2c_write", ret, IOTBUS_ERROR_NONE, iotbus_i2c_stop(h_i2c));
+
+	ret = iotbus_i2c_read(h_i2c, read_buf, 1);
+	TC_ASSERT_GT_CLEANUP("iotbus_i2c_read", ret, IOTBUS_ERROR_NONE, iotbus_i2c_stop(h_i2c));
+
+	ret = iotbus_i2c_write(h_i2c, &write_cmd[1], 1);
+	TC_ASSERT_GT_CLEANUP("iotbus_i2c_write", ret, IOTBUS_ERROR_NONE, iotbus_i2c_stop(h_i2c));
 
-	ret = iotbus_i2c_read(h_i2c, szReadBuf, 10);
+	ret = iotbus_i2c_read(h_i2c, read_buf, 1);
```

Comment:
You didn't check read value so that there is no difference between first and second. Why do you try it twice?